jax/experimental/jax2tf/jax2tf.py

```diff
@@ -1113,6 +1113,29 @@ def func1(*jax_args):
 
 tf_impl[lax.scan_p] = _scan
 
+def _top_k(operand: TfVal, k: int) -> Tuple[TfVal, TfVal]:
+  # Some types originally incompatible with tf.math.top_k can be promoted
+  # to a compatible type without loss of precision.
+  def promote_dtype(dtype):
+    if dtype in [jnp.bool_, jnp.uint8, jnp.uint16]:
+      return jnp.uint32
+    if dtype in [jnp.int8, jnp.int16]:
+      return jnp.int32
+    if dtype in [jnp.float16]:
+      return jnp.float32
+    if dtype in [jnp.bfloat16]:
+      return tf.bfloat16
```

Comment:
I am wondering why we return here `tf.bfloat16`, and in all other branches we returns a `jnp` type?

Reply:
Addressed in latest commits.